### PR TITLE
Merge wrench estimation offset removal code

### PR DESCRIPTION
**Pull request by Yeshasvitvs**
This PR merges the recent commits did for offset removal of external wrench estimation to the main branch of  [feature/visualize-berdy-estimated-wrench](https://github.com/robotology/human-dynamics-estimation/tree/feature/visualize-berdy-estimated-wrench). 

@lrapetti @claudia-lat @DanielePucci

### CODE REVIEW

**Comment by Yeshasvitvs** [comment]
@lrapetti please squash the commits you see relevant. Thanks.

**Comment by lrapetti** [comment]
Main improvement with this PR:
- Enable running and visualising multiple HDE at the same time by adding:
  - portPrefix option for `HumanDynamicEstimator` (https://github.com/robotology/human-dynamics-estimation/pull/156/commits/08ff46bf7d6ceccc65863c5e882213fb744a821e).
  - using prefix also for the HumanDynamicsPublisher ros node (https://github.com/robotology/human-dynamics-estimation/pull/156/commits/6d85ec9467005b8d71fef8c1c21ca059b11f76dc)

  the configuration files for running multiple HDE are also added (`Human_1.xml` and `Human_2.xml`)
- Add an option to select the offset removal methods (two of which are implemented in the same PR) https://github.com/robotology/human-dynamics-estimation/pull/156/commits/cf03f9217099c653f97d59ff55f896572a3092ef. The option can be se to:
  - `source`: is the offset removal previously implemented that remove the offset equally on the sensor wrench sources by sending `removeWrenchOffset` signal. 
  - `source-dynamic`: the dynamic offset removal can set the two feet standing offset with `removeWrenchOffset`, as `source` and this offset will be used when standing on two feet. When instead moving on a single support a different offset is used, and it is zero, or can be set with `removeWrenchOffset <wrench source name>` (e.g. standing on right foot and sending the command `removeWrenchOffset RightFoot`). The usage of two feet or single foot support offset is defined in "fuzzy logic fashion" using gaussian membership function to identify foot in contact.
  - `model`: offset is directly removed from the model as a ficticious wrench at the com (so doesn't change the wrench sensor measurements).
  
  In the latest test we have been using `source-dynamic` without actually removing the offset on single support. Eventually in the future we may converge to a single option after proper testing.
- A command for fixing the estiamted external wrenches has been implemented https://github.com/robotology/human-dynamics-estimation/pull/156/commits/b6862fb54b7588a3f54b7f188bcdc3623c2f6d58. This is triggered by the command `fixEstimatedWrenches`, and can be removed with `resetEstimatedWrenches`.
- It has been introduced a `wrenchEstimationForceLowerThreshold ` option that introduce a lower treshold for the estimated wrenches (below which are set to zero) https://github.com/robotology/human-dynamics-estimation/pull/156/commits/e1e392b66fb9f98b67f98cf0571064ad9400bda7
- other smaller fixies and options (https://github.com/robotology/human-dynamics-estimation/pull/156/commits/778d9a5d21e867de2a01bda39a305e9a0d3ae1fe https://github.com/robotology/human-dynamics-estimation/pull/156/commits/aafdc3fce6087ac846c50d85f3bbdbd1bb250cd6 https://github.com/robotology/human-dynamics-estimation/pull/156/commits/76b5ae23211be34ee7daeafcfa7a945dca159085 https://github.com/robotology/human-dynamics-estimation/pull/156/commits/c9aed86677bb9258f5521b3fe15843b172a3524f)

**Comment by lrapetti** [comment]
> > Screenshot from 2019-11-20 12-13-24
> > @lrapetti please add a new line character `\n` at the end of each command for clear visibility of the available options through `help` command.

done https://github.com/robotology/human-dynamics-estimation/pull/156/commits/18be95067c0ea8fd3ad5610f234e81cedb58bd87

File: conf/xml/Human_1.xml [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+            <paramlist name="networks">
+                <elem name="HumanDynamicsEstimatorLabel1">HumanStateProvider_1</elem>
+                <elem name="HumanDynamicsEstimatorLabel2">HumanWrenchProvider_1</elem>
+            </paramlist>
+        </action>
+        <action phase="shutdown" level="5" type="detach"/>
+    </device>
+
+    <device type="analogServer" name="HumanEstimatedWrenchWrapper_1">
+        <param name="name">/HDE_2/HumanEstimatedWrenchWrapper/wrench:o</param>
```
[the remark this answers is not in the document]
Comment (Yeshasvitvs) [reply conf/xml/Human_1.xml]: Done in https://github.com/robotology/human-dynamics-estimation/pull/156/commits/9ee18b161252208b490f897d2744ffdf6af7aa50

File: conf/xml/Human_1.xml [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+        <param name="useDirectBaseMeasurement">false</param>
+        <!-- optimization parameters -->
+        <param name="maxIterationsIK">300</param>
+        <param name="ikLinearSolver">ma27</param>
+        <param name="ikPoolSizeOption">2</param>
+        <param name="posTargetWeight">0.0</param>
+        <param name="rotTargetWeight">1.0</param>
+        <param name="costRegularization">0.000001</param>
+        <param name="costTolerance">0.001</param>
+        <param name="rpcPortPrefix">first</param>
```
[the remark this answers is not in the document]
Comment (Yeshasvitvs) [reply conf/xml/Human_1.xml]: Done in https://github.com/robotology/human-dynamics-estimation/pull/156/commits/9ee18b161252208b490f897d2744ffdf6af7aa50

File: conf/xml/Human_2.xml [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+    </device>
+
+    <device type="human_dynamics_estimator" name="HumanDynamicsEstimator_2">
+        <param name="period">0.100</param>
+        <param name="urdf">humanSubject01_66dof.urdf</param>
+        <param name="baseLink">Pelvis</param>
+        <param name="number_of_wrench_sensors">4</param>
+        <param name="wrench_sensors_link_name">(LeftFoot RightFoot LeftHand RightHand)</param>
+        <param name="comConstraintIncludeAllLinks">false</param>
+        <param name="rpcPortPrefix">second</param>
```
[the remark this answers is not in the document]
Comment (Yeshasvitvs) [reply conf/xml/Human_2.xml]: Done in https://github.com/robotology/human-dynamics-estimation/pull/156/commits/9ee18b161252208b490f897d2744ffdf6af7aa50

File: conf/xml/Human_2.xml [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+        <param name="useDirectBaseMeasurement">false</param>
+        <!-- optimization parameters -->
+        <param name="maxIterationsIK">300</param>
+        <param name="ikLinearSolver">ma27</param>
+        <param name="ikPoolSizeOption">2</param>
+        <param name="posTargetWeight">0.0</param>
+        <param name="rotTargetWeight">1.0</param>
+        <param name="costRegularization">0.000001</param>
+        <param name="costTolerance">0.001</param>
+        <param name="rpcPortPrefix">second</param>
```
[the remark this answers is not in the document]
Comment (Yeshasvitvs) [reply conf/xml/Human_2.xml]: Done in https://github.com/robotology/human-dynamics-estimation/pull/156/commits/9ee18b161252208b490f897d2744ffdf6af7aa50

File: conf/xml/Human_1.xml [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+    </device>
+
+    <device type="human_dynamics_estimator" name="HumanDynamicsEstimator_1">
+        <param name="period">0.100</param>
+        <param name="urdf">humanSubject01_66dof.urdf</param>
+        <param name="baseLink">Pelvis</param>
+        <param name="number_of_wrench_sensors">4</param>
+        <param name="wrench_sensors_link_name">(LeftFoot RightFoot LeftHand RightHand)</param>
+        <param name="comConstraintIncludeAllLinks">false</param>
+        <param name="rpcPortPrefix">first</param>
```
[the remark this answers is not in the document]
Comment (Yeshasvitvs) [reply conf/xml/Human_1.xml]: Done in https://github.com/robotology/human-dynamics-estimation/pull/156/commits/9ee18b161252208b490f897d2744ffdf6af7aa50

**Comment by Yeshasvitvs** [comment]
As discussed with @lrapetti we will proceed to merge this PR and keep all the changes related to HDE use of stack of tasks BERDY in [feature/visualize-berdy-estimated-wrench](https://github.com/robotology/human-dynamics-estimation/tree/feature/visualize-berdy-estimated-wrench) branch.

**Pull request merged**